include/mimalloc-internal.h

```diff
@@ -158,8 +158,10 @@ static inline bool mi_mul_overflow(size_t count, size_t size, size_t* total) {
 #if __has_builtin(__builtin_umul_overflow) || __GNUC__ >= 5
 #if (MI_INTPTR_SIZE == 4)
   return __builtin_umul_overflow(count, size, total);
-#else
+#elif ULONG_MAX != 0xffffffffl
```

Comment:
Can we use `(INTPTR_MAX == LONG_MAX)` instead?

Reply:
I see that you already used the `SIZE_MAX == ULONG_MAX` test, which I think is splendid. Better than what I had.